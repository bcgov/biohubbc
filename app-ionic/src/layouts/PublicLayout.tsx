import { CssBaseline, makeStyles, Theme } from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles((theme: Theme) => ({
  publicLayoutRoot: {
    height: 'inherit',
    width: 'inherit',
    display: 'flex',
    flexDirection: 'column'
  },
  mainContent: {
    flex: 1,
    width: 'inherit',
    height: 'inherit',
    overflow: 'auto'
  },
  container: {
    paddingTop: theme.spacing(4),
    paddingBottom: theme.spacing(4)
  }
}));

const PublicLayout: React.FC = (props) => {
  const classes = useStyles();

  return (
    <div className={classes.publicLayoutRoot}>
      <CssBaseline />
      <main className={classes.mainContent}>
        {/* <ErrorBoundary FallbackComponent={ErrorDialog}> */}
        {React.Children.map(props.children, (child: any) => {
          return React.cloneElement(child, { classes: classes });
        })}
        {/* </ErrorBoundary> */}
      </main>
    </div>
  );
};

export default PublicLayout;
