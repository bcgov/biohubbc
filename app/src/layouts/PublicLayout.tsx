import { Box, CssBaseline, makeStyles, Theme } from '@material-ui/core';
import Footer from 'components/layout/Footer';
import Header from 'components/layout/Header';
import React from 'react';

const useStyles = makeStyles((theme: Theme) => ({
  mainContent: {
    flex: 1,
    width: 'inherit',
    height: 'inherit',
    overflow: 'auto'
  },
  container: {
    paddingTop: theme.spacing(4),
    paddingBottom: theme.spacing(4)
  }
}));

const PublicLayout: React.FC = (props) => {
  const classes = useStyles();

  return (
    <Box mb={2} height="inherit" width="inherit" display="flex" flexDirection="column">
      <CssBaseline />

      <Header />

      <main className={classes.mainContent}>
        {React.Children.map(props.children, (child: any) => {
          return React.cloneElement(child, { classes: classes });
        })}
      </main>

      <footer>
        <Footer />
      </footer>
    </Box>
  );
};

export default PublicLayout;
