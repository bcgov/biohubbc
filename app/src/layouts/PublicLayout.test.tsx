import { render } from '@testing-library/react';
import React from 'react';
import PublicLayout from './PublicLayout';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

const history = createMemoryHistory();

describe('PublicLayout', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Router history={history}>
        <PublicLayout>
          <div>
            <p>The public layout content</p>
          </div>
        </PublicLayout>
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
