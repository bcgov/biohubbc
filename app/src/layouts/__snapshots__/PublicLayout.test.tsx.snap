// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicLayout renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      style="box-shadow: none;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular app-header MuiToolbar-gutters"
      >
        <a
          aria-label="Go to BioHub Home"
          class="brand"
          href="/projects"
        >
          <picture>
            <source
              media="(min-width: 1200px)"
              srcset="gov-bc-logo-horiz.png"
            />
            <source
              media="(min-width: 600px)"
              srcset="gov-bc-logo-vert.png"
            />
            <img
              alt="Government of British Columbia"
              src="gov-bc-logo-vert.png"
            />
          </picture>
          BioHub
        </a>
      </div>
      <div
        aria-label="Main Navigation"
        class="MuiToolbar-root MuiToolbar-dense main-nav MuiToolbar-gutters"
        role="navigation"
      >
        <a
          color="inherit"
          href="/projects"
        >
          Projects
        </a>
      </div>
    </header>
    <main
      class="makeStyles-mainContent-1"
    >
      <div
        classes="[object Object]"
      >
        <p>
          The public layout content
        </p>
      </div>
    </main>
    <footer
      class="app-footer"
    >
      <div
        aria-label="Footer"
        class="MuiToolbar-root MuiToolbar-regular app-footer-toolbar MuiToolbar-gutters"
        role="navigation"
      >
        <ul>
          <li>
            <a
              href="https://www.gov.bc.ca/gov/content/home/disclaimer"
            >
              Disclaimer
            </a>
          </li>
          <li>
            <a
              href="https://www.gov.bc.ca/gov/content/home/privacy"
            >
              Privacy
            </a>
          </li>
          <li>
            <a
              href="https://www.gov.bc.ca/gov/content/home/accessible-government"
            >
              Accessibility
            </a>
          </li>
          <li>
            <a
              href="https://www.gov.bc.ca/gov/content/home/copyright"
            >
              Copyright
            </a>
          </li>
        </ul>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
